File: source forum/ressource_diapo/fibo.c
#include <stdio.h>

// fonction recursive
int fibonacci_recursive(int n) {
    if (n <= 1) {
        return n;
    }
    return fibonacci_recursive(n - 1) + fibonacci_recursive(n - 2);
}

// fonction iterative
int fibonacci_iterative(int n) {
    int a = 0;
    int b = 1;
    int c;
    for (int i = 0; i < n; i++) {
        c = a + b;
        a = b;
        b = c;
    }
    return a;
}

File: source forum/ressource_diapo/fibo.ml
(* fonction recursive *)
let rec fibonacci_recursive n =
  match n with
  | 0 -> 0
  | 1 -> 1
  | _ -> fibonacci_recursive (n - 1) + fibonacci_recursive (n - 2)

(* fonction iterative *)
let fibonacci_iterative n =
  let a = ref 0 in
  let b = ref 1 in
  for i = 0 to n - 1 do
    let c = !a + !b in
    a := !b;
    b := c
  done;
